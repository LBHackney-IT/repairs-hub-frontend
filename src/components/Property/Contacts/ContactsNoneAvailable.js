const ContactsNoneAvailable = () => {
  return (
    <div className="govuk-warning-text">
      <span className="govuk-warning-text__icon" aria-hidden="true">
        !
      </span>
      <strong className="govuk-warning-text__text">
        <span className="govuk-warning-text__assistive">Warning</span>
        No contact details available for this property
      </strong>
    </div>
  )
}

export default ContactsNoneAvailable
