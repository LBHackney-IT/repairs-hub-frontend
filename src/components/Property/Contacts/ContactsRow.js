import PropTypes from 'prop-types'

const Row = ({ contact }) => {
  return (
    <tr className="govuk-table__row">
      <td className="govuk-table__cell">
        {[contact?.firstName, contact?.lastName].join(' ')}
      </td>
      <td className="govuk-table__cell">{contact.phoneNumbers[0]}</td>
      <td className="govuk-table__cell">{contact.phoneNumbers[1]}</td>
      <td className="govuk-table__cell">{contact.phoneNumbers[2]}</td>
    </tr>
  )
}

Row.propTypes = {
  contact: PropTypes.shape({
    firstName: PropTypes.string,
    lastName: PropTypes.string,
    phoneNumbers: PropTypes.arrayOf(PropTypes.string),
  }).isRequired,
}

export default Row
