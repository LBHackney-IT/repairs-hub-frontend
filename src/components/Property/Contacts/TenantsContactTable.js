import { useState } from 'react'
import ConfirmationModal from '../../ConfirmationModal'

const TenantHeading = ({ fullName, personId }) => {
  const domain = 'https://manage-my-home-development.hackney.gov.uk'

  return (
    <div className="tenantContactsTable-heading govuk-body">
      <h4>{fullName}</h4>

      <a
        className="tenantContactsTable-button"
        href={`
      ${domain}/person/${personId}/edit-contact-details
      `}
        target="_blank"
      >
        Edit contact details
      </a>
    </div>
  )
}

const TenantPhoneNumber = ({ phoneNumber, fullName }) => {
  const [showRemoveModal, setShowRemoveModal] = useState(false)
  const [showMainContactModal, setShowMainContactModal] = useState(false)

  return (
    <li className="tenantContactsTable-phoneNumberGrid govuk-body">
      <div className="tenantContactsTable-valueGroup">
        <h4 className="tenantContactsTable-phoneSubType">
          {phoneNumber.subType}
        </h4>
        <div className="tenantContactsTable-phone">{phoneNumber.value}</div>
      </div>
      {!!phoneNumber?.description && (
        <div className="tenantContactsTable-description">
          {phoneNumber.description}
        </div>
      )}
      <div className="tenantContactsTable-buttonGroup">
        <button
          type="button"
          className="tenantContactsTable-button"
          onClick={() => setShowMainContactModal(true)}
        >
          Set as main contact
        </button>

        <button
          type="button"
          className="tenantContactsTable-button"
          onClick={() => setShowRemoveModal(true)}
        >
          Remove
        </button>
      </div>

      <ConfirmationModal
        title={'Set as main contact'}
        showDialog={showMainContactModal}
        setShowDialog={setShowMainContactModal}
        modalText={
          <p className="govuk-body">
            Are you sure you want to change the contact type from{' '}
            <strong style={{ textTransform: 'capitalize' }}>
              {phoneNumber.subType}
            </strong>{' '}
            to{' '}
            <strong style={{ textTransform: 'capitalize' }}>
              Main contact
            </strong>
            ?
          </p>
        }
        onSubmit={() => {
          setShowMainContactModal(false)
          alert('onsubmit')
        }}
        yesButtonText={'Set as main contact'}
      />

      <ConfirmationModal
        title={'Remove contact details'}
        showDialog={showRemoveModal}
        setShowDialog={setShowRemoveModal}
        modalText={
          <p className="govuk-body">
            Are you sure you want to permanently remove{' '}
            <strong
              style={{ textTransform: 'capitalize', whiteSpace: 'nowrap' }}
            >
              {phoneNumber.subType}: {phoneNumber.value}
            </strong>{' '}
            from {fullName}?
          </p>
        }
        onSubmit={() => {
          setShowRemoveModal(false)
          alert('onsubmit')
        }}
        yesButtonText={'Remove phone number'}
      />
    </li>
  )
}

const TenantPhoneNumberList = ({ phoneNumbers, fullName }) => {
  return (
    <ul>
      {phoneNumbers.map((x) => (
        <TenantPhoneNumber phoneNumber={x} fullName={fullName} />
      ))}
    </ul>
  )
}

const TenantContact = ({ tenant }) => {
  return (
    <li className="tenantContactsTable-contact">
      <TenantHeading
        fullName={tenant.fullName}
        personId={'a597c0d4-e182-2db8-8003-bc6486356414'}
      />

      <hr />

      {tenant.phoneNumbers.length === 0 && <p>No contact details</p>}

      <TenantPhoneNumberList
        phoneNumbers={tenant.phoneNumbers}
        fullName={tenant.fullName}
      />
    </li>
  )
}

const TenantContactsTable = ({ tenants }) => {
  return (
    <div className="tenantContactsTable">
      <ul>
        {tenants.map((x) => (
          <TenantContact tenant={x} />
        ))}
      </ul>
    </div>
  )
}

export default TenantContactsTable
