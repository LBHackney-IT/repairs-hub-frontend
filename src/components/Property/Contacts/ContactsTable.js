import PropTypes from 'prop-types'
import ContactsRow from './ContactsRow'

const ContactsTable = ({ contacts }) => {
  return (
    <table className="govuk-table lbh-table">
      <caption className="govuk-table__caption">Contacts</caption>
      <thead className="govuk-table__head">
        <tr className="govuk-table__row">
          <th className="govuk-table__header" scope="col">
            Name
          </th>
          <th className="govuk-table__header" scope="col">
            Telephone 1
          </th>
          <th className="govuk-table__header" scope="col">
            Telephone 2
          </th>
          <th className="govuk-table__header" scope="col">
            Telephone 3
          </th>
        </tr>
      </thead>
      <tbody className="govuk-table__body">
        {contacts.map((contact, index) => (
          <ContactsRow key={index} contact={contact} />
        ))}
      </tbody>
    </table>
  )
}

ContactsTable.propTypes = {
  contacts: PropTypes.arrayOf(PropTypes.object).isRequired,
}

export default ContactsTable
