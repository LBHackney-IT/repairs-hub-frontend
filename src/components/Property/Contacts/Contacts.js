import PropTypes from 'prop-types'
import WarningText from '../../WarningText'
import ContactsTable from './ContactsTable'

const Contacts = ({ contacts }) => {
  return contacts.length < 1 ? (
    <WarningText text="No contact details available for this property" />
  ) : (
    <ContactsTable contacts={contacts} />
  )
}

Contacts.propTypes = {
  contacts: PropTypes.arrayOf(PropTypes.object).isRequired,
}

export default Contacts
