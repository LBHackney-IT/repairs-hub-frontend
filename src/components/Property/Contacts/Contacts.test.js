import { render } from '@testing-library/react'
import Contacts from './Contacts'

describe('Contacts component', () => {
  describe('when supplied with an empty contacts list', () => {
    const contacts = []

    it('renders text indicating that no contacts are available', () => {
      const { asFragment } = render(<Contacts contacts={contacts} />)

      expect(asFragment()).toMatchSnapshot()
    })
  })

  describe('when supplied with a list of contacts', () => {
    const contacts = [
      {
        firstName: 'Mark',
        lastName: 'Gardner',
        phoneNumbers: ['00000111111', '00000222222', '00000333333'],
      },
      {
        firstName: 'Luam',
        lastName: 'Berhane',
        phoneNumbers: ['', '', '00000333333'],
      },
    ]

    it('renders a table with a row for each contact', () => {
      const { asFragment } = render(<Contacts contacts={contacts} />)

      expect(asFragment()).toMatchSnapshot()
    })
  })
})
