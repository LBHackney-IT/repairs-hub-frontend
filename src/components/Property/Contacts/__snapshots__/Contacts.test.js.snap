// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts component when supplied with a list of contacts renders a table with a row for each contact 1`] = `
<DocumentFragment>
  <table
    class="govuk-table"
  >
    <caption
      class="govuk-table__caption"
    >
      Contacts
    </caption>
    <thead
      class="govuk-table__head"
    >
      <tr
        class="govuk-table__row"
      >
        <th
          class="govuk-table__header"
          scope="col"
        >
          Name
        </th>
        <th
          class="govuk-table__header"
          scope="col"
        >
          Telephone 1
        </th>
        <th
          class="govuk-table__header"
          scope="col"
        >
          Telephone 2
        </th>
        <th
          class="govuk-table__header"
          scope="col"
        >
          Telephone 3
        </th>
      </tr>
    </thead>
    <tbody
      class="govuk-table__body"
    >
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell"
        >
          Mark Gardner
        </td>
        <td
          class="govuk-table__cell"
        >
          00000111111
        </td>
        <td
          class="govuk-table__cell"
        >
          00000222222
        </td>
        <td
          class="govuk-table__cell"
        >
          00000333333
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell"
        >
          Luam Berhane
        </td>
        <td
          class="govuk-table__cell"
        />
        <td
          class="govuk-table__cell"
        />
        <td
          class="govuk-table__cell"
        >
          00000333333
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`Contacts component when supplied with an empty contacts list renders text indicating that no contacts are available 1`] = `
<DocumentFragment>
  <div
    class="govuk-warning-text"
  >
    <span
      aria-hidden="true"
      class="govuk-warning-text__icon"
    >
      !
    </span>
    <strong
      class="govuk-warning-text__text"
    >
      <span
        class="govuk-warning-text__assistive"
      >
        Warning
      </span>
      No contact details available for this property
    </strong>
  </div>
</DocumentFragment>
`;
